tests/test_connect.py

```diff
@@ -878,6 +905,33 @@ def tearDown(self):
         finally:
             await con.close()
 
+    async def test_ssl_connection_sslmode(self):
+        async def verify_works(dsn):
+            con = None
+            try:
+                con = await self.connect(dsn)
+                self.assertEqual(await con.fetchval('SELECT 42'), 42)
+            finally:
+                if con:
+                    await con.close()
+
+        async def verify_fails(dsn):
+            con = None
+            try:
+                with self.assertRaises(asyncpg.PostgresError):
+                    await self.connect(dsn)
+                    await con.fetchval('SELECT 42')
+            finally:
+                if con:
+                    await con.close()
+
+        await verify_fails('postgresql://ssl_user@localhost/?sslmode=disable')
+        await verify_works('postgresql://ssl_user@localhost/?sslmode=allow')
+        await verify_works('postgresql://ssl_user@localhost/?sslmode=prefer')
+        await verify_works('postgresql://ssl_user@localhost/?sslmode=require')
+        await verify_fails('postgresql://ssl_user@localhost/?sslmode=verify-ca')
```

Comment:
I think it's important to also test that `verify-ca` and `verify-full` are working (both success and failure in the presence of an invalid server certificate).  For that we need to be able to parse `sslrootcert` from the DSN (and set it to `SSL_CA_CERT_FILE`).

Reply:
Added a test for that. Wasn't via adding the `sslrootcert` param, but, that shouldn't be too hard for anyone who wants to.